src/v/pandaproxy/schema_registry/seq_writer.cc

```diff
@@ -0,0 +1,129 @@
+//// Copyright 2021 Vectorized, Inc.
+////
+//// Use of this software is governed by the Business Source License
+//// included in the file licenses/BSL.md
+////
+//// As of the Change Date specified in that file, in accordance with
+//// the Business Source License, use of this software will be governed
+//// by the Apache License, Version 2.
+
+#include "pandaproxy/schema_registry/seq_writer.h"
+
+#include "pandaproxy/error.h"
+#include "pandaproxy/logger.h"
+#include "pandaproxy/schema_registry/client_fetch_batch_reader.h"
+#include "pandaproxy/schema_registry/storage.h"
+#include "random/simple_time_jitter.h"
+#include "vlog.h"
+
+#include <seastar/core/coroutine.hh>
+
+using namespace std::chrono_literals;
+
+namespace pandaproxy::schema_registry {
+
+// TODO: hook consume_to_store into advance_offset
+
+/// Call this before reading from the store, if servicing
+/// a REST API endpoint that requires global knowledge of latest
+/// data (i.e. any listings)
+ss::future<> seq_writer::read_sync() {
+    auto offsets = co_await _client.local().list_offsets(
+      model::schema_registry_internal_tp);
+
+    const auto& topics = offsets.data.topics;
+    if (topics.size() != 1 || topics.front().partitions.size() != 1) {
+        auto ec = kafka::error_code::unknown_topic_or_partition;
+        throw kafka::exception(ec, make_error_code(ec).message());
+    }
+
+    const auto& partition = topics.front().partitions.front();
+    if (partition.error_code != kafka::error_code::none) {
+        auto ec = partition.error_code;
+        throw kafka::exception(ec, make_error_code(ec).message());
+    }
+
+    co_await wait_for(partition.offset - model::offset{1});
+}
+
+ss::future<> seq_writer::wait_for(model::offset offset) {
+    auto fn = [offset](seq_writer& seq) {
+        auto sem_units = ss::get_units(seq._wait_for_sem, 1);
```

Comment:
Does this need to be held of the duration of the operation? It might need moving into a finally continuation on `client_fetch_batch_reader::consume`?